docs/source/testing.rst

```diff
@@ -0,0 +1,218 @@
+.. _testing:
+
+===========
+GUI Testing
+===========
+
+GUI testing involves a slew of difficulties on top of what is normally
+encountered when writing typical unit tests. This is a result of having to deal
+with the underlying event loop that drives the GUI. Consider a scenario in
+which you have a simple :class:`~traits.has_traits.HasTraits` class and you call
+:meth:`~traits.has_traits.HasTraits.configure_traits` on it to start the
+application. Typically python execution stops at this point and control is
+passed to the underlying event loop. In a test context though, we need
+direct access to that event loop. To simulate and test GUI interactions we need
+to be able to post events to the event loop, process them manually, and then
+make assertions about the desired behavior. Even further, the event loop is
+global state. Therefor, great care needs to be taken in each test to pick up
+after itself to avoid interactions with other tests. This is still necessary
+even if the test fails. 
+
+Pyface provides a few utilities that are useful in this process.  Namely,
+:class:`~pyface.util.GuiTestAssistant` and
+:class:`~pyface.util.ModalDialogTester`.
+
+
+GuiTestAssistant
+================
+
+.. note::
+
+   GuiTestAssistant is currently only available on Qt.
+
+:class:`GuiTestAssistant` is a mixin class intended to to augment
+:class:`python:unittest.TestCase`. In fact, it has
+:class:`~traits.testing.unittest_tools.UnittestTools` from traits as a base
+class and thus gives access to methods like
```

Comment:
personally, i would prefer "In fact, it inherits from ..." instead of the current "... from traits as a base class ..."

Reply:
It took me a couple of goes to understand that the "traits" in "from traits" was a reference to the Traits library (rather than a reference to traited attributes). Capitalising "Traits" might help here.